src/pudl/output/eia.py

```diff
@@ -145,20 +144,71 @@ def _out_eia__yearly_generators(
     )
 
     # If any generator data is completely empty, drop it.
-    # These are three known generators that originate from harvesting the plant and
+    # These are a handful of known generators that originate from harvesting the plant and
     # generator IDs found in the plant_id_eia_direct_support_x and
     # generator_id_direct_support_x in EIA 860 energy storage tables, in
     # order to enable foreign key relationships with these columns.
     # They do not show up in any other tables and thus lack data in all columns.
     # For more, see issue #3695 and PR #3699.
     empty_generator_ids = [
-        (9170, "3093", "2023-01-01"),
-        (34516, "SOL1", "2023-01-01"),
-        (64966, "GEN1", "2023-01-01"),
+        (1, "CFCPV"),
+        (1, "HB2PV"),
+        (1, "LUNPV"),
+        (1, "MIDPV"),
+        (1, "RDYPV"),
+        (1, "TE1PV"),
+        (9170, "3093"),
+        (34516, "SOL1"),
+        (56951, "VESTA"),
+        (57991, "PV2"),
+        (58644, "ALL"),
+        (62355, "2WPSO"),
+        (62760, "SONRI"),
+        (62975, "SYNLB"),
+        (63210, "SAINT"),
+        (69312, "EcheB"),
+        (64094, "PBS0L"),
+        (64182, "PRAPV"),
+        (64245, "90FIB"),
+        (64246, "99MTB"),
+        (64436, "WLB"),
+        (64876, "OHAMP"),
+        (64966, "GEN1"),
+        (64996, "ARCPV"),
+        (65084, "ELDPV"),
+        (66574, "USFL"),
+        (65647, "GEN1"),
+        (65859, "GEN03"),
+        (66502, "FWSOL"),
+        (66394, "GEN01"),
+        (66502, "FWSOL"),
+        (66897, "ALBPV"),
+        (67744, "RS1"),
+        (68239, "C234B"),
+        (69312, "ECHEB"),
     ]
-    out_df = drop_all_null_records_with_multiindex(
-        out_df, ["plant_id_eia", "generator_id", "report_date"], empty_generator_ids
-    )
+
+    non_null_cols = [
+        "report_date",
+        "plant_id_eia",
+        "generator_id",
+        "fuel_type_count",
+    ]  # Columns that will have values
+    all_null_gens = out_df[
+        out_df[[col for col in out_df.columns if col not in non_null_cols]]
+        .isnull()
+        .all(axis=1)
+    ]
+    all_null_gens_ids = [
+        (int(row.plant_id_eia), row.generator_id) for row in all_null_gens.itertuples()
+    ]
+    unexpected_gens = set(all_null_gens_ids) - set(empty_generator_ids)
```

Comment:
non-blocking but you could make `empty_generator_ids` a pd.Index and use the `all_null_gens.index.difference(empty_generator_ids)` so you don't have to do the itertuples situation. i think it would be easier to read but totally not blocking this looks like it does what you want!